smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/integration/http-binding-utils.ts

```diff
@@ -1,7 +1,2 @@
 const isSerializableHeaderValue = (value: any): boolean =>
-  value !== undefined &&
-  value !== null &&
-  value !== "" &&
-  (!Object.getOwnPropertyNames(value).includes("length") ||
-    value.length != 0) &&
-  (!Object.getOwnPropertyNames(value).includes("size") || value.size != 0);
+  value != undefined && value != null;
```

Comment:
`value != null` is enough, since `undefined` will match

Reply:
is it possible to cover this with some unit tests? would be nice to see what cases we are considering

Reply:
added unit tests